include/swift/AST/DiagnosticConsumer.h

```diff
@@ -111,6 +112,55 @@ class NullDiagnosticConsumer : public DiagnosticConsumer {
                         ArrayRef<DiagnosticArgument> FormatArgs,
                         const DiagnosticInfo &Info) override;
 };
+
+/// \brief DiagnosticConsumer that funnels diagnostics in certain files to
+/// particular sub-consumers.
+///
+/// Diagnostics that are not in one of the special files are emitted into every
```

Comment:
Adding a sentence or two to explain what a "certain" or "special" file is would help a novice to understand what's going on here. What's the basic problem? Something like: Xcode needs serialized diagnostic files with the diagnostic routed to the right file. [Why?] This bigger picture is the sort of thing one cannot get from just reading the code.

Reply:
I was doing that somewhat intentionally, since this is in theory a reusable component. But you're right, a few lines on the example, intended use would make sense.